Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -331,6 +463,189 @@ def flyTo(self, pathPointIndex):
         self.cameraNode.EndModify(wasModified)
         self.cameraNode.ResetClippingRange()
 
+class EndoscopyLogic:
+    """Direct the editing of control points.
+    Enables us to edit given control points, the "cursor" is an MRML node
+    that is over the selected control point and indicates the orienation of the camera
+    at that node. Currently a vtkMRMLMarkupsPlane is used. When the user selects
+    a new control point the plane is moved to that.
+    """
+
+    # TODO
+    # observer plane editing
+    # update data in controlpoint from rotation
+    # Verify if world or local axes on the plane are correct
+
+    def __init__(self):
+        self.n = 0
+        self.positions = []
+        self.orientations = []
+        # self.fov = []
+        self.curveNode = None
+        self.curveNodeObserverTags = []
+
+        self.cursorNode = None
+        self.cursorNodeObserverTags = []
+
+        self.currentIndex = -1
+
+    def createOrientations(self, count):
+        quaternions = []
+        for axes in self.orientations:
+            quaternions.append(toQuaternion(axes[0], axes[1], axes[2]))
+
+        # TODO NEXT !!!
```

Comment:
@jadh4v  This is where progress is atm, to calcluate a correct list of orientations we need to know how the control points are spaced out